src/lower/lower_indexexpr_taco.cpp

```diff
@@ -0,0 +1,1168 @@
+
+    // Let the old lowerIndexExpression pass haneld expressions taco can't
+    Stmt lowerIndexExpression(Stmt s) {
+      Func f("", {}, {}, s, environment);
+      f.setStorage(storage);
+      f = lowerIndexExpressions(f);
+      s = f.getBody();
+      s = removeVarDecls(s).first;
+      storage = f.getStorage();
+      return s;
+    }
+
+    void handleAssignLikeStmt(Stmt op, ScalarType componentType) {
+      struct : public IRVisitor {
+        bool value = false;
+
+        virtual void visit(const IndexExpr* op) {
+          value = true;
+          IRVisitor::visit(op);
+        }
+
+        virtual void visit(const IndexedTensor* op) {
+          value = true;
+          IRVisitor::visit(op);
+        }
+      } hasIndexVar;
+
+      op.accept(&hasIndexVar);
+      if (hasIndexVar.value) {
+        if (tacoCannotHandle(op)) {
+          // taco can't handle transpose yet
+          stmt = lowerIndexExpression(op);
+        } else {
+          try {
+            stmt = TacoLower().rewrite(op, componentType, &storage);
+          } catch (const TacoException& e) {
+            std::cerr << "Warning: taco cannot handle (unknown reason)\n"
+                      << util::toString(stmt) << "\n";
+            stmt = lowerIndexExpression(op);
+          }
+          stmt = Comment::make(util::toString(op), stmt, false, true);
+        }
+      } else {
+        stmt = op;
+      }
+    }
+
+    virtual void visit(const AssignStmt* op) {
+      handleAssignLikeStmt(op, op->value.type().toTensor()->getComponentType());
+    }
+
+    virtual void visit(const TensorWrite* op) {
+      handleAssignLikeStmt(op, op->value.type().toTensor()->getComponentType());
+    }
+
+    virtual void visit(const FieldWrite* op) {
+      handleAssignLikeStmt(op, op->value.type().toTensor()->getComponentType());
+    }
+
+    virtual void visit(const Func* op) {
+      storage = op->getStorage();
+      environment =op->getEnvironment();
+      IRRewriter::visit(op);
+      func.setStorage(storage);
+    }
+
+
+    bool tacoCannotHandle(Stmt stmt) {
+#define CHECK(x, msg) if (x) {\
+    std::cerr << "Warning: taco cannot handle " msg "\n"\
+              << util::toString(stmt) << "\n";\
+    return true;\
+  }
+
+      CHECK(isTranspose(stmt), "transpose");
+      CHECK(isSelfAssignmentWithDifferentIndices(stmt),
+            "self assignment with different indices");
+      CHECK(sparsityMismatch(stmt),
+            "same index var but different sparsity");
+      CHECK(isSparseFollowedByDense(stmt),
+            "sparse dimension followed by dense dimension");
+      CHECK(storageKindUnsupported(stmt, storage),
+            "diagonal or stencil storage");
+      CHECK(hasFixedIndexVar(stmt), "fixed index expression");
+      CHECK(hasDivision(stmt), "division");
+      return false;
+    }
+  } rewriter;
+
+  func = rewriter.rewrite(func);
+  func = insertVarDecls(func);
+  func = insertInitializations(func); /// @todo needed due to bugs in taco
```

Comment:
Is this still needed with the latest version of taco? I believe it should now generate output initialization code as well.